resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/KubernetesClientUtils.scala

```diff
@@ -90,29 +92,67 @@ private[spark] object KubernetesClientUtils extends Logging {
       .build()
   }
 
-  private def loadSparkConfDirFiles(conf: SparkConf): Map[String, String] = {
+  private def orderFilesBySize(confFiles: Seq[File]): Seq[File] = {
+    val fileToFileSizePairs = confFiles.map(f => (f, f.getName.length + f.length()))
+    // sort first by name and then by length, so that during tests we have consistent results.
+    fileToFileSizePairs.sortBy(f => f._1).sortBy(f => f._2).map(_._1)
```

Comment:
Ur, this has a bug because we will consider file name later by `truncatedMapSize = truncatedMapSize + (fileName.length + fileContent.length)`.
```scala
scala> Seq("b" -> 1, "abcdef" -> 1).sortBy(f => f._1).sortBy(f => f._2)
res14: Seq[(String, Int)] = List((abcdef,1), (b,1))

scala> Seq("b" -> 1, "abcdef" -> 1).sortBy(f => f._1).sortBy(f => f._2).map(f => f._1.length + f._2)
res15: Seq[Int] = List(7, 2)
```

Reply:
Values are actually sum total of the size of filename and it's content. Two items with same size would mean, that if first one did not fit, next one won't fit either. Anyway, I have addressed it.